dex/order/order.go

```diff
@@ -153,23 +153,41 @@ type Order interface {
 	// Quote returns the unique integer identifier of the quote asset as defined
 	// in the asset package.
 	Quote() uint32
+
+	// IsSell indicates the the order is selling the base asset (false indicates
+	// selling the quote asset). This helps identify the asset of the backing
+	// coins returned by Coins(). Note that a cancel order will return false.
+	IsSell() bool
+
+	// Coins returns the backing coins of either base or quote asset depending
+	// on IsSell.
+	//
+	// TODO!!! The orders must be updated to track the current backing coins,
+	// not just the original backing coins. This is critical for partially
+	// filled orders where each fill creates change that must then be tracked as
+	// the new backing coins.
```

Comment:
This is a big TODO.  We discussed this in matrix recently, but in summary, the Order needs to know the coins that are currently used to back the order.  This can change because of partial fills.

Say an order is partially filled and it stays on the books.  Now say the swap succeeds and the backing coins are spent.   From this swap that partially filled the order, there was change.  That change would need to be tracked now as it would be used to back the order with the new/reduced remaining amount.

Reply:
I don't think we should try to track the change output with the coin locker. The changes in #91 left a `TxID` method to hopefully deal with change outputs. See the [use of `(MarketTunnel).TxMonitored` by the `OrderRouter`](https://github.com/decred/dcrdex/blob/b64658ef0d9fdcc9222182e94519d643e9195fee/server/market/orderrouter.go#L520). I was guessing that the "dex-monitored" change output can be checked by asking the `Swapper` whether it is currently tracking the transaction ID. The `Swapper` now [saves the `Coins`](https://github.com/decred/dcrdex/blob/b64658ef0d9fdcc9222182e94519d643e9195fee/server/swap/swap.go#L63-L68) involved in the swap, so it should be straightforward to implement.

Implementation of coin tracking through the locker would necessitate the user submitting the change coin information with the `init` and `redeem` route payloads, and probably some messy edge cases. Notably, right now, the exchange wallet interface from #72 and #92 don't even give the user information about the coin output. It's tracked internally.

Reply:
So it seems the swapStatus.coins needs to be used in the Swapper's coin locker?  I think it will be clear to you what needs to be done there when you get a look at the swap.go changes.

Reply:
As it stands, I think this PR is going to fail to lock or double lock partial filled orders.

Reply:
> The changes in #91 left a `TxID` method to hopefully deal with change outputs. See the [use of `(MarketTunnel).TxMonitored` by the `OrderRouter`](https://github.com/decred/dcrdex/blob/b64658ef0d9fdcc9222182e94519d643e9195fee/server/market/orderrouter.go#L520). I was guessing that the "dex-monitored" change output can be checked by asking the `Swapper` whether it is currently tracking the transaction ID. The `Swapper` now [saves the `Coins`](https://github.com/decred/dcrdex/blob/b64658ef0d9fdcc9222182e94519d643e9195fee/server/swap/swap.go#L63-L68) involved in the swap, so it should be straightforward to implement.

Is the following diff what you had in mind? https://github.com/decred/dcrdex/pull/95/files#diff-5f1d20749025f318cd1a8ba0915ebd37R418-R456